### ISSUE TITLE
pl.rank_gene_groups using gene_symbols key word not working

### ISSUE THREAD

**Issue by Xparx** [scverse/scanpy#1758]
Hello Team,
```
sc.__version__
'1.7.0rc2.dev6+g5fc12f4a'
```
I just pulled the latest master to see if the issue was just in my earlier version.

I'm trying to plot rank_gene_groups dot plot 
`sc.pl.rank_genes_groups_dotplot(adata, n_genes=4, values_to_plot='logfoldchanges', cmap='bwr', gene_symbols='Uniq_Name', use_raw=False)`

However it doesn't work with the following tranceback:
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-43-4ca50e495f54> in <module>
----> 1 sc.pl.rank_genes_groups_dotplot(adata, n_genes=4, values_to_plot='logfoldchanges', cmap='bwr', gene_symbols='Uniq_Name', use_raw=False)  # swap_axes=True)

~/projects/scanpy/scanpy/plotting/_tools/__init__.py in rank_genes_groups_dotplot(adata, groups, n_genes, groupby, values_to_plot, min_logfoldchange, key, show, save, return_fig, **kwds)
    656     """
    657 
--> 658     return _rank_genes_groups_plot(
    659         adata,
    660         plot_type='dotplot',

~/projects/scanpy/scanpy/plotting/_tools/__init__.py in _rank_genes_groups_plot(adata, plot_type, groups, n_genes, groupby, values_to_plot, min_logfoldchange, key, show, save, return_fig, **kwds)
    435             from .._dotplot import dotplot
    436 
--> 437             _pl = dotplot(
    438                 adata,
    439                 var_names,

~/projects/scanpy/scanpy/plotting/_dotplot.py in dotplot(adata, var_names, groupby, use_raw, log, num_categories, expression_cutoff, mean_only_expressed, cmap, dot_max, dot_min, standard_scale, smallest_dot, title, colorbar_title, size_title, figsize, dendrogram, gene_symbols, var_group_positions, var_group_labels, var_group_rotation, layer, swap_axes, dot_color_df, show, save, ax, return_fig, vmin, vmax, vcenter, norm, **kwds)
    940         del kwds['color_map']
    941 
--> 942     dp = DotPlot(
    943         adata,
    944         var_names,

~/projects/scanpy/scanpy/plotting/_dotplot.py in __init__(self, adata, var_names, groupby, use_raw, log, num_categories, categories_order, title, figsize, gene_symbols, var_group_positions, var_group_labels, var_group_rotation, layer, expression_cutoff, mean_only_expressed, standard_scale, dot_color_df, dot_size_df, ax, vmin, vmax, vcenter, norm, **kwds)
    130         **kwds,
    131     ):
--> 132         BasePlot.__init__(
    133             self,
    134             adata,

~/projects/scanpy/scanpy/plotting/_baseplot_class.py in __init__(self, adata, var_names, groupby, use_raw, log, num_categories, categories_order, title, figsize, gene_symbols, var_group_positions, var_group_labels, var_group_rotation, layer, ax, vmin, vmax, vcenter, norm, **kwds)
    109         self._update_var_groups()
    110 
--> 111         self.categories, self.obs_tidy = _prepare_dataframe(
    112             adata,
    113             self.var_names,

~/projects/scanpy/scanpy/plotting/_anndata.py in _prepare_dataframe(adata, var_names, groupby, use_raw, log, num_categories, layer, gene_symbols)
   1864         groupby.remove(groupby_index)
   1865     keys = list(groupby) + list(np.unique(var_names))
-> 1866     obs_tidy = get.obs_df(
   1867         adata, keys=keys, layer=layer, use_raw=use_raw, gene_symbols=gene_symbols
   1868     )

~/projects/scanpy/scanpy/get.py in obs_df(adata, keys, obsm_keys, layer, gene_symbols, use_raw)

KeyError: "Could not find keys '['KY.Chr1.1190', ..., 'KY.Chr9.987']' in columns of `adata.obs` or in gene_symbols column `adata.var[Uniq_Name].values`."
```

I cant see how to use gene_symbols in the `tl.rank_gene_groups` either. Am I missing something?

**Comment by ivirshup** [comment]
This seems mostly fine. I would definitely suggest updating to a more recent version, as there could definitely be issues in pre-release builds.

If that doesn't solve your problem, could you confirm if `"KY.Chr1.1190" in adata.var["Uniq_Name"]`?

**Comment by Xparx** [comment]
```
sc.__version__
'1.8.0.dev78+gc488909a'
```

It seems to be working but I'm currently on a different dataset. What I noticed was that if I didn't have the same ID columns in my `adata.var` when setting adata.raw I couldn't use `gene_symbols`. After setting `adata.var` so it had the same IDs before setting `adata.raw` made it possible. 
In other words if adata.raw was missing the notation it failed for me (different error though). 
I will give an update when I get back to the dataset above. Just to make sure it's the same issue.

**Comment by ivirshup** [comment]
> What I noticed was that if I didn't have the same ID columns in my adata.var when setting adata.raw I couldn't use gene_symbols.

Ah, if you're using the values in `raw` for differential expression the column used for `gene_symbols` should be in `raw.var` this is because `raw` can have a different set of variables than the main object. So this case, at least, is expected.

**Comment by Xparx** [comment]
Getting back to this. I still have the same issue as before.

For some reason this does not work on my data. I don't have any `raw` set at all in this data.
```
sc.tl.rank_genes_groups(adata, 'celltypes', method='t-test')
sc.pl.rank_genes_groups_matrixplot(adata, n_genes=5, cmap='viridis', gene_symbols='Uniq_Name')
```

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-41-3b246f8b6bcc> in <module>
----> 1 sc.pl.rank_genes_groups_matrixplot(adata, n_genes=5, use_raw=False, cmap='viridis', gene_symbols='Uniq_Name')

~/projects/scanpy/scanpy/plotting/_tools/__init__.py in rank_genes_groups_matrixplot(adata, groups, n_genes, groupby, values_to_plot, min_logfoldchange, key, show, save, return_fig, **kwds)
    823     """
    824 
--> 825     return _rank_genes_groups_plot(
    826         adata,
    827         plot_type='matrixplot',

~/projects/scanpy/scanpy/plotting/_tools/__init__.py in _rank_genes_groups_plot(adata, plot_type, groups, n_genes, groupby, values_to_plot, min_logfoldchange, key, show, save, return_fig, **kwds)
    448             from .._matrixplot import matrixplot
    449 
--> 450             _pl = matrixplot(
    451                 adata, var_names, groupby, values_df=values_df, return_fig=True, **kwds
    452             )

~/projects/scanpy/scanpy/plotting/_matrixplot.py in matrixplot(adata, var_names, groupby, use_raw, log, num_categories, figsize, dendrogram, title, cmap, colorbar_title, gene_symbols, var_group_positions, var_group_labels, var_group_rotation, layer, standard_scale, values_df, swap_axes, show, save, ax, return_fig, vmin, vmax, vcenter, norm, **kwds)
    345     """
    346 
--> 347     mp = MatrixPlot(
    348         adata,
    349         var_names,

~/projects/scanpy/scanpy/plotting/_matrixplot.py in __init__(self, adata, var_names, groupby, use_raw, log, num_categories, categories_order, title, figsize, gene_symbols, var_group_positions, var_group_labels, var_group_rotation, layer, standard_scale, ax, values_df, vmin, vmax, vcenter, norm, **kwds)
    109         **kwds,
    110     ):
--> 111         BasePlot.__init__(
    112             self,
    113             adata,

~/projects/scanpy/scanpy/plotting/_baseplot_class.py in __init__(self, adata, var_names, groupby, use_raw, log, num_categories, categories_order, title, figsize, gene_symbols, var_group_positions, var_group_labels, var_group_rotation, layer, ax, vmin, vmax, vcenter, norm, **kwds)
    109         self._update_var_groups()
    110 
--> 111         self.categories, self.obs_tidy = _prepare_dataframe(
    112             adata,
    113             self.var_names,

~/projects/scanpy/scanpy/plotting/_anndata.py in _prepare_dataframe(adata, var_names, groupby, use_raw, log, num_categories, layer, gene_symbols)
   1864         groupby.remove(groupby_index)
   1865     keys = list(groupby) + list(np.unique(var_names))
-> 1866     obs_tidy = get.obs_df(
   1867         adata, keys=keys, layer=layer, use_raw=use_raw, gene_symbols=gene_symbols
   1868     )

~/projects/scanpy/scanpy/get/get.py in obs_df(adata, keys, obsm_keys, layer, gene_symbols, use_raw)
    270         alias_index = None
    271 
--> 272     obs_cols, var_idx_keys, var_symbols = _check_indices(
    273         adata.obs,
    274         var.index,

~/projects/scanpy/scanpy/get/get.py in _check_indices(dim_df, alt_index, dim, keys, alias_index, use_raw)
    165             not_found.append(key)
    166     if len(not_found) > 0:
--> 167         raise KeyError(
    168             f"Could not find keys '{not_found}' in columns of `adata.{dim}` or in"
    169             f" {alt_repr}.{alt_search_repr}."

KeyError: "Could not find keys '['KY.Chr1.1388', 'KY.Chr1.1475', 'KY.Chr1.2070', 'KY.Chr1.2214', 'KY.Chr1.2297', 'KY.Chr1.686', 'KY.Chr11.93', 'KY.Chr13.413', 'KY.Chr2.1171', 'KY.Chr2.1545', 'KY.Chr3.405', 'KY.Chr4.246', 'KY.Chr4.468', 'KY.Chr5.382', 'KY.Chr7.503', 'KY.Chr7.846', 'KY.Chr8.1168', 'KY.Chr8.473', 'KY.Chr9.813', 'KY.Chr9.912']' in columns of `adata.obs` or in adata.var['Uniq_Name']."
```
The KeyError is odd. `adata.var['Uniq_Name']` Contains the actual gene identifiers. The ones in `KeyError` is in `adata.var_names`. So it looks like it searches for the var_names in Uniq_Name for some reason.

I pulled from this repo just now.

**Comment by Xparx** [comment]
Surprisingly `sc.pl.rank_genes_groups(adata, n_genes=25, sharey=False, gene_symbols='Uniq_Name')` works just fine.

**Comment by ivirshup** [comment]
Ah, I think this is a bit of a known issue with a fix in the works here: #1529. Essentially there is a bug in some of the `rank_genes_groups_*` functions with how they handle `gene_symbols`.

**Comment by ivirshup** [comment]
Closing to consolidate discussion to #1796
